blockchain/contracts/DungeonDefenders/DefenderUtils.sol

```diff
@@ -0,0 +1,44 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "./DefenderFactory.sol";
+
+contract DefenderUtils is DefenderFactory {
+    event DefenderLeveledUp(uint256 defenderId, uint8 level);
+
+    mapping (uint256 => uint256) public defenderExperience;
+
+    // based on exp(level) = round(2^((level - 1) / 3) * 1000 - 1000)
+    uint32[10] levelToExp = [
+        0,      // level 1
+        260,    // level 2
+        587,    // level 3
+        1000,   // level 4
+        1520,   // level 5
+        2175,   // level 6
+        3000,   // level 7
+        4040,   // level 8
+        5849,   // level 9
+        7000    // level 10
+    ];
+
+    function gainExperience(uint256 _tokenId, uint256 _expToGain) public {
```

Comment:
function should not be callable by everyone obviously, but everything looking really nice :)

Reply:
Yea was thinking of adding permissions and safe math in a pr later